Content.Server/Traits/Assorted/EmotionLoopComponent.cs

```diff
@@ -0,0 +1,21 @@
+using System.Numerics;
+
+namespace Content.Server.Traits.Assorted;
+
+/// <summary>
+/// This component allows triggering any emotion at random intervals.
+/// </summary>
+[RegisterComponent, Access(typeof(EmotionLoopSystem))]
+public sealed partial class EmotionLoopComponent : Component
+{
+    /// <summary>
+    /// A random interval between emotions (minimum, maximum).
+    /// </summary>
+    [DataField("timeBetweenEmotions", required: true)]
+    public Vector2 TimeBetweenEmotions { get; private set; }
```

Comment:
I think it may be better to split this into two fields rather than forcing the two together as a vector.